--- config/config.sub	2020-03-07 11:02:26.365690000 +0100
+++ config/config.sub	2020-03-07 11:08:39.419117166 +0100
@@ -312,7 +312,7 @@
 	| ubicom32 \
 	| v850 | v850e | v850e1 | v850e2 | v850es | v850e2v3 \
 	| visium \
-	| wasm32 \
+	| wasm32 | wasm64 \
 	| x86 | xc16x | xstormy16 | xtensa \
 	| z8k | z80)
 		basic_machine=$basic_machine-unknown
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd | -amdhsa* | -unleashed* | -emscripten* | -wasi*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
